src/Symfony/Component/Translation/TranslatableMessage.php

```diff
@@ -52,6 +53,13 @@ public function getDomain(): ?string
 
     public function trans(TranslatorInterface $translator, string $locale = null): string
     {
-        return $translator->trans($this->getMessage(), $this->getParameters(), $this->getDomain(), $locale);
+        $parameters = array_map(function($parameter) use($translator, $locale) {
+            if ($parameter instanceof TranslatableInterface) {
```

Comment:
IMHO this should be handled by contract, by any translator

Reply:
@ro0NL Ok so I remove this part and change all implementations of `TranslatorInterface`?